### ISSUE TITLE
My legacy fontforge→LaTeX script has stopped working.

### ISSUE THREAD

**Issue by Anaphory** [fontforge/fontforge#2863]
Long time ago, I wrote a script that would take a fontforge `.sfdir` directory and generate a LaTeX font from it, because the `tfm` file generated by fontforge contained only minor information on ligatures, special substitutions etc. [That file](https://github.com/Anaphory/Jena1330/blob/master/BuildAndInstall) is part of my Jena1330 package. The core functionality is a small-ish [python script](https://github.com/Anaphory/Jena1330/blob/master/mkligtable.py) which reads the properties file in the sfdir and generates a `pl` file from it.

I just wanted to generate a new LaTeX release of Jena1330, but the `pltotf` step of my script fails now, due to character values above 255 in the ligature table.
1. Is this workaround even still necessary, or does the current development version of fontforge generate clean `tfm`s that include substitution information?
2. Assuming there is no easy replacement for my workaround, and given that my font does contain less than 256 different characters, how do I convince fontforge to stick the characters all in encoding slots below 255? A simple `Reencode("compacted");` before the pfb export did not seem to help.
3. If that part of LaTeX/pfb generation is still missing, how should I restructure my `mkligtable.py` to be useful to the fontforge project?

**Comment by jtanx** [comment]
Hmm, I'm not too sure. I've looked at the code, and there certainly does seem to be a function that adds ligatures to TFM files ([TfmAddLiga](https://github.com/fontforge/fontforge/blame/a2a3d0377e886888afdebb811041bf548b761f2c/fontforge/splinesaveafm.c#L2611)) - this was added around 2006. So if you tried before 2006 and it didn't work, give it another shot - if you tried after 2006, well I guess something must be broken with the generation.

Wrt. encoding slots, you'll have to play around with the reencode options - but not the compact option; this one is really only meant for the gui to remove unused slots from the display.

**Comment by Anaphory** [comment]
It was definitely around 2010 that I started this project, so I guess there will be some use in helping to fix TfmAddLiga.

Some of my characters are alternative forms of basic characters that don't appear in common encodings (word-initial w with a swash to the left, for example) and I think they got pushed outside the 256 range recently because I added some new accented characters in the upper range. What might me a good way to make them accessible to LaTeX?

(Incidentally, is there a mailing list I should post things like this to instead?)
